src/cljs/athens/electron.cljs

```diff
@@ -323,6 +325,40 @@
   ;; open or create a new starter db
 
   ;; Watch filesystem, e.g. in case db is updated via Dropbox sync
+  
+    
+  ;; Update index.transit    
+  (defn write-db-index [file filepath]
+    (let [w  (.. fs (writeFile filepath file (fn [e] (dispatch [:boot/desktop]))))])
+  )  
+  
+  (defn open-dialog-index
+    "Allow user to open a Backup file."
+    [filepath]
+    (js/alert "Your data file is corrupted or incorrect, please select a backup file.")
+    (let [res  (.showOpenDialogSync dialog (clj->js {:properties ["openFile"]
+                                                     :filters    [{:name "Transit" :extensions ["bkp"]}]}))
+          open-file (first res)]
+      (when (and open-file (.existsSync fs open-file))
+        (let [read-db (.readFileSync fs open-file)
+              db-file      (try  (dt/read-transit-str read-db)(catch  :default e (open-dialog-index filepath)))              
+              ] 
```

Comment:
these 3 lines are duplicated with lines 352-354, consider making a helper function to deduplicate

Reply:
could you explain about the helper function thanks